### PR TITLE
Improvements to Instance.Parent documentation

### PR DESCRIPTION
**Pull request by EmeraldSlash**
## Changes

Added a common piece of undocumented knowledge: that changing properties while an instance is parented can be expensive and trigger more work.

e.g. https://devforum.roblox.com/t/psa-dont-use-instancenew-with-parent-argument/30296

But also based on my observations from using the engine and from looking at the microprofiler and seeing things like GUI layouting work being performed after reading/changing a property.

Also improved the wording of the Object Replication section so that it is clearer, as I couldn't understand what it is was saying the first time I read it.

## Checks

By submitting your pull request for review, you agree to the following:

- [X] This contribution was created in whole or in part by me, and I have the right to submit it under the terms of this repository's open source licenses.
- [X] I understand and agree that this contribution and a record of it are public, maintained indefinitely, and may be redistributed under the terms of this repository's open source licenses.
- [X] To the best of my knowledge, all proposed changes are accurate.

### CODE REVIEW

**Comment by EmeraldSlash** [comment]
> Hey @EmeraldSlash, it seems like this is covered pretty well in the Object Replication section right below the new paragraph. Is there some additional nuance here that I'm missing?

That section only refers to replication. But there are other cases which have nothing to do with replication where parenting order is important and where changing properties can incur additional work. So that's why I think it's a good idea to document that behaviour more generally.

I agree as it's written now there's a bit of duplication.. I wanted to keep the changes minimal and not rewrite the whole section or remove emphasis on the replication part which is still important.

File: content/en-us/reference/engine/classes/Instance.yaml [hunk 1]
```diff
@@ -191,9 +191,16 @@ properties:
       Instance.new("NumberValue", workspace)
       ```
 
+      Whena a change is made to certain properties while an instance is
```
Comment (aetter) [inline content/en-us/reference/engine/classes/Instance.yaml:194]:
```suggestion
      When a change is made to certain properties while an instance is
```

File: content/en-us/reference/engine/classes/Instance.yaml [hunk 2]
```diff
@@ -191,9 +191,16 @@ properties:
...
+      parented in the `Class.DataModel` the engine may need to perform extra
```
Comment (aetter) [inline content/en-us/reference/engine/classes/Instance.yaml:195]:
```suggestion
      parented in the `Class.DataModel`, the engine may need to perform extra
```

File: content/en-us/reference/engine/classes/Instance.yaml [hunk 3]
```diff
@@ -191,9 +191,16 @@ properties:
...
+      work internally (for things like replication, rendering, and GUI
+      layouting). Therefore where possible it is a good idea to change an
```
Comment (aetter) [inline content/en-us/reference/engine/classes/Instance.yaml:197]:
```suggestion
      layout). Whenever possible, change an
```

**Review by aetter — APPROVED** [review]

**Review by aetter — APPROVED** [review]

**Comment by aetter** [comment]
Sounds good, thanks. I might give it a pass later to try and deduplicate, but that's no reason not to take the improvement now. Thanks again!

**Pull request merged**